### PR TITLE
ED-3042 feat : [ED-Portal]: Generic implementation of Hardcoded BMGS 

### PR DESCRIPTION
**Pull request by princegupta1131**
# SunbirdEd - Portal
---

### Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
---

#### Example
- [x] Applicable
- [ ] Not applicable

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/src/app/modules/core/services/user/user.service.ts [hunk 1]
```diff
@@ -499,7 +503,15 @@ export class UserService {
...
     }));
   }
-
+  
+  /**
+    * @description - This method is called in the case where either of onboarding or framework popup is disabled and setGuest value is made true
+  */
+  setGuestUser(value: boolean, defaultFormatedName: string): void{
+    this.setGuest = value;
+    this.formatedName = defaultFormatedName;
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/core/services/user/user.service.ts:512]: @mithun30052001 check the variable name

File: src/app/client/src/app/modules/public/services/csl-framework/csl-framework.service.ts [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+ *                          rejects if an error occurs during the operation
+ */
+  public setFWCatConfigFromCsl(userSelFramework: any): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      if (userSelFramework) {
+        this.csFrameworkService.getFrameworkConfigMap(userSelFramework, {
+          apiPath: '/api/framework/v1/'
+        }).subscribe(
+          (fwData) => {
+            console.log('getFrameworkConfigMap success', fwData);
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/public/services/csl-framework/csl-framework.service.ts:138]: remove console

File: src/app/client/src/app/modules/public/services/csl-framework/csl-framework.service.ts [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+            localStorage.removeItem('fwCategoryObject');
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/public/services/csl-framework/csl-framework.service.ts:139]: use cacheservice for localstorage bindings

File: src/app/client/src/app/modules/content-search/components/search-filter/search-filter.component.ts [hunk 4]
```diff
@@ -54,10 +57,10 @@ export class SearchFilterComponent implements OnInit, OnDestroy {
     private contentSearchService: ContentSearchService,
     private activatedRoute: ActivatedRoute, private cdr: ChangeDetectorRef,
     public layoutService: LayoutService, private formService: FormService,
-    private cacheService: CacheService, private utilService: UtilService) { }
+    private cacheService: CacheService, private utilService: UtilService, private cslFrameworkService: CslFrameworkService ) { }
 
   get filterData() {
-    return _.get(this.pageData, 'metaData.filters') || ['medium', 'gradeLevel', 'board', 'channel', 'subject', 'audience', 'publisher', 'se_subjects', 'se_boards', 'se_gradeLevels', 'se_mediums'];
+    return _.get(this.pageData, 'metaData.filters') || [ this.frameworkCategories?.fwCategory1?.code,this.frameworkCategories?.fwCategory2?.code,this.frameworkCategories?.fwCategory3?.code,this.frameworkCategories?.fwCategory4?.code, 'channel', 'audience', 'publisher', 'se_subjects', 'se_boards', 'se_gradeLevels', 'se_mediums'];
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/content-search/components/search-filter/search-filter.component.ts:63]: make this optimal using method getallfwcatname() created in csl service

File: src/app/client/src/app/modules/content-search/services/content-search/content-search.service.ts [hunk 5]
```diff
@@ -24,12 +25,19 @@ export class ContentSearchService {
...
   get searchResults$(): Observable<any[]> {
     return this._searchResults$.asObservable()
       .pipe(skipWhile(data => data === undefined || data === null));
   }
 
-  constructor(private frameworkService: FrameworkService, private channelService: ChannelService) { }
+  constructor(private frameworkService: FrameworkService, private channelService: ChannelService, private cslFrameworkService:CslFrameworkService) { 
+    this.frameworkCategories = this.cslFrameworkService.getFrameworkCategories();
+    this.frameworkCategoriesObject = this.cslFrameworkService.getFrameworkCategoriesObject();
+   requiredCategories = { categories: `${this.frameworkCategories?.fwCategory1?.code,this.frameworkCategories?.fwCategory2?.code,this.frameworkCategories?.fwCategory3?.code,this.frameworkCategories?.fwCategory4?.code}}` };
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/content-search/services/content-search/content-search.service.ts:39]: optimize the code

File: src/app/client/src/app/modules/core/services/search/search.service.ts [hunk 6]
```diff
@@ -286,17 +293,19 @@ export class SearchService {
   * @param {option}
   **/
   public updateOption(option: any) {
-    if (_.get(option, 'data.request.filters.board')) {
-      option.data.request.filters['se_boards'] = option.data.request.filters.board;
-      delete option.data.request.filters.board;
+    this.globalFilterCategories = this.cslFrameworkService.getAlternativeCodeForFilter();
+    console.log('updateOption', this.globalFilterCategories);
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/core/services/search/search.service.ts:297]: remove console

File: src/app/client/src/app/modules/core/services/search/search.service.ts
Comment (abhishekpnt) [inline src/app/client/src/app/modules/core/services/search/search.service.ts:1]: optimize the keys

File: src/app/client/src/app/modules/core/services/search/search.service.ts [hunk 7]
```diff
@@ -498,70 +507,41 @@ export class SearchService {
...
-          facet['label'] = _.get(this.resourceService, 'frmelmnts.lbl.orgname');
-          facet['placeholder'] =  _.get(this.resourceService, 'frmelmnts.lbl.orgname');
-          facet['values'] = _.map(facet.values || [], value => ({ ...value, name: value.orgName }));
-          break;
-          case 'additionalCategories':
-            facet['index'] = '71';
-            facet['label'] = this.resourceService.frmelmnts.lbl.additionalCategories;
-            facet['placeholder'] = this.resourceService.frmelmnts.lbl.selectAdditionalCategory;
+          // case 'audience':
+          //   facet['placeholder'] = this.resourceService.frmelmnts.lbl.selectMeantFor;
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/core/services/search/search.service.ts:537]: cleanup the comments

File: src/app/client/src/app/modules/public/services/csl-framework/csl-framework.service.ts [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+      if (fwCode && typeCheck && contentData[fwCode]) {
+        const keyValueObj = {
+          labels: fwData?.label,
+          value: contentData[fwCode],
+          index: fwData?.index
+        };
+        result.push(keyValueObj);
+      }
+    });
+    console.log('contentData', result);
```
Comment (abhishekpnt) [inline src/app/client/src/app/modules/public/services/csl-framework/csl-framework.service.ts:121]: remove console

**Review by rajeevsathish — APPROVED** [review]

**Pull request merged**